wrangler-service/src/main/java/io/cdap/wrangler/service/directive/AbstractDirectiveHandler.java

```diff
@@ -97,6 +107,76 @@ public void destroy() {
     return new CommonDirectiveExecutor(contextProvider, composite).executeDirectives(namespace, directives, sample);
   }
 
+
+  /**
+   * Formats Structure and Array records in a "pseudojson" in which it collapses
+   * complex types by callint toString().  This is done so the complex types are shown
+   * in a similar way inside and outside the structure records.
+   */
+  private static class WranglerDisplaySerializer implements TypeAdapterFactory {
+    @Override
+    public <T> TypeAdapter<T> create(Gson gson, TypeToken<T> type) {
+      // Let GSON Handle primitives and collections
+      if (ClassUtils.isPrimitiveOrWrapper(type.getRawType())
+          || (Iterable.class.isAssignableFrom(type.getRawType()))) {
+        return null;
+      }
+      if (Row.class.isAssignableFrom(type.getRawType())) {
+        return (TypeAdapter<T>) rowClassAdapter(gson);
+      }
+
+      return (TypeAdapter<T>) defaultClassAdapter(gson);
+    }
+
+    private TypeAdapter<Row> rowClassAdapter(Gson gson) {
+      return new TypeAdapter<Row>() {
+        @Override
+        public void write(JsonWriter out, Row value) throws IOException {
+          if (value == null) {
+            return;
+          }
+          final TypeAdapter<Object> elementAdapter = gson.getAdapter(Object.class);
```

Comment:
nit: move this into `rowClassAdapter` method. This would ensure it's called once and not for each record, similar to `defaultClassAdapter`